src/modules/common/components/Selector.js

```diff
@@ -1,37 +1,52 @@
+// @flow
+
 import React from 'react'
-import PropTypes from 'prop-types'
 
 import SelectorItemModel from '../models/SelectorItemModel'
-import { ActiveElement, Element, Wrapper } from './Selector.styles'
+import { SelectedElement, ActiveElement, Wrapper, InactiveElement } from './Selector.styles'
+import { translate } from 'react-i18next'
+import type { I18nTranslate } from '../../../flowTypes'
+import Tooltip from './Tooltip'
+
+type Props = {
+  verticalLayout: boolean,
+  closeDropDownCallback?: () => {},
+  items: Array<SelectorItemModel>,
+  activeItemCode?: string,
+  tooltip: string,
+  t: I18nTranslate
+}
 
 /**
  * Displays a Selector showing different items
  */
-class Selector extends React.Component {
-  static propTypes = {
-    verticalLayout: PropTypes.bool,
-    closeDropDownCallback: PropTypes.func,
-    items: PropTypes.arrayOf(PropTypes.instanceOf(SelectorItemModel)).isRequired,
-    /** The code of the item which is currently active **/
-    activeItemCode: PropTypes.string
-  }
-
+export class Selector extends React.Component<Props> {
   getItems () {
-    return this.props.items.map(item => {
-      if (item.code === this.props.activeItemCode) {
+    const {items, activeItemCode, closeDropDownCallback, tooltip, t} = this.props
+    return items.map(item => {
+      if (item.code === activeItemCode) {
+        return (
+          <SelectedElement key={item.code}
+                         onClick={closeDropDownCallback}>
+            {item.name}
+          </SelectedElement>
+        )
+      } else if (item.href) {
         return (
           <ActiveElement key={item.code}
-                         onClick={this.props.closeDropDownCallback}>
+                onClick={closeDropDownCallback}
+                to={item.href}>
             {item.name}
           </ActiveElement>
         )
       } else {
         return (
-          <Element key={item.code}
-                onClick={this.props.closeDropDownCallback}
-                to={item.href}>
-            {item.name}
-          </Element>
+          <span data-tip={t(tooltip)} key={item.code}>
+            <InactiveElement>
+              {item.name}
+            </InactiveElement>
+            <Tooltip />
```

Comment:
We shouldn't spawn multiple ReactTooltip "instances" (this can lead to unexpected z-index behavior - i've had this problem 😅 ). Instead we should call `ReactTooltip.rebuild()` in `componentDidRender()` and - if needed - in `componentDidUpdate()`.

Reply:
Ahh okay, only componentDidRender did not work, componentDidUpdate solved it :+1: thanks